stl/inc/flat_set

```diff
@@ -100,12 +95,13 @@ public:
     _Base_flat_set(_Tsorted _Tsort, const container_type& _Cont, const key_compare& _Comp, const _Alloc& _Al)
         : _Base_flat_set(_Tsort, container_type(_Cont, _Al), _Comp) {}
 
-    explicit _Base_flat_set(const key_compare& _Comp) : _My_pair(_One_then_variadic_args_t{}, _Comp) {}
+    explicit _Base_flat_set(const key_compare& _Comp) : _Mycont(), _Mycomp(_Comp) {}
     template <_Allocator_for<container_type> _Alloc>
-    _Base_flat_set(const key_compare& _Comp, const _Alloc& _Al) : _My_pair(_One_then_variadic_args_t{}, _Comp, _Al) {}
+    _Base_flat_set(const key_compare& _Comp, const _Alloc& _Al) : _Mycont(_Al), _Mycomp(_Comp) {}
     template <_Allocator_for<container_type> _Alloc>
-    explicit _Base_flat_set(const _Alloc& _Al) : _My_pair(_Zero_then_variadic_args_t{}, _Al) {}
+    explicit _Base_flat_set(const _Alloc& _Al) : _Mycont(_Al), _Mycomp() {}
 
+    // TRANSITION, an allocator-aware container may not support "C(_First, _Last, _Al)".

```

Comment:
I think there is no need to use `make_obj_using_allocator` for `C(another,al) and C(al)`, as these expressions are required for [allocator-aware containers](https://eel.is/c++draft/container.alloc.reqmts).

For `C(first,last,al)`, `C(from_range...,al)` and `C(ilist,al)`, however, I realize that `make_obj_using_allocator` is not capable. As a failure to construct with these arguments are considered ill-formed by `make_obj_using_allocator`.
> https://eel.is/c++draft/allocator.uses.construction#5.4.sentence-1
> Otherwise, the program is ill-formed.

However, in `flat_set` these constructors only say the container should be constructed with the allocator, then behave as if inserting the range. So these constructors should be valid even if the container doesn't support these constructions.

(The most conservative fix would be fall back to constructing the container using only `_Al`, then doing insertion in the function body.)